multisense_ros/include/multisense_ros/camera.h

```diff
@@ -333,17 +333,20 @@ class Camera {
     //
     // Diagnostics
     diagnostic_updater::Updater diagnostic_updater_;
-    void deviceInfoDiagnostic(diagnostic_updater::DiagnosticStatusWrapper &stat);
-    void deviceStatusDiagnostic(diagnostic_updater::DiagnosticStatusWrapper &stat);
+    void deviceInfoDiagnostic(diagnostic_updater::DiagnosticStatusWrapper& stat);
+    void deviceStatusDiagnostic(diagnostic_updater::DiagnosticStatusWrapper& stat);
+    void ptpStatusDiagnostic(diagnostic_updater::DiagnosticStatusWrapper& stat);
 
-    void diagnosticTimerCallback(const ros::TimerEvent &);
+    void diagnosticTimerCallback(const ros::TimerEvent&);
     ros::Timer diagnostic_trigger_;
 
     //
-    // Timesync settings
+    // Timestamping and timesync settings
+    ros::Time convertImageTimestamp(uint32_t time_secs, uint32_t time_microsecs);
```

Comment:
might make sense to name what you're converting to - like imageTimestampToRosTime() or something